src/libflate/lib.rs

```diff
@@ -35,13 +34,27 @@
 extern crate libc;
 
 use libc::{c_void, size_t, c_int};
+use std::fmt;
 use std::ops::Deref;
 use std::ptr::Unique;
 use std::slice;
 
+#[derive(Clone, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct Error(());
+
+impl fmt::Debug for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        "decompression error".fmt(f)
+    }
+}
+
+impl Drop for Error {
```

Comment:
This is a private library to rustc where we do not need to worry about backwards compatibility. Additionally breaking `let Error(_) = foo` does not seem to me to warrant an extra blank impl of `Drop`. Having this extra impl is unfortunate because it is unexpected and especially without a comment it will be difficult for later readers to understand why this is here.